server/src/main/scala/org/apache/spark/sql/crossdata/session/HazelcastSessionProvider.scala

```diff
@@ -44,7 +50,25 @@ class HazelcastSessionProvider( @transient sc: SparkContext,
   import HazelcastSessionProvider._
   import XDSharedState._
 
-  private val sharedState = new XDSharedState(sc, Option(userConfig))// TODO add sqlConf
+  catalogConfig = userConfig.getConfig(CoreConfig.CatalogConfigKey)
```

Comment:
userConfig is comming of server-reference instead of core-reference and it will not find catalog proerties